iib/workers/tasks/legacy.py

```diff
@@ -0,0 +1,226 @@
+
+    :param str package_dir: path to the exported package directory
+    :param str from_index: the pull specification of the image image
+    :raises iib.exceptions.IIBError: if the generated package info is missing
+    """
+    package_name = package_dir.split('/download-', 1)[-1]
+    log.info('Verifying package_name %s', package_name)
+    # opm does not fail when the package is missing in the index image, hence we
+    # check the number of generated files. If it's equal to 1, that means only an empty
+    # `package.yaml` file is generated and the package is missing
+    if len(os.listdir(package_dir)) == 1:
+        raise IIBError(f'package {package_name} is missing in index image {from_index}')
+
+    try:
+        output = api.build_and_verify(source_dir=package_dir)
+        if not output.is_valid:
+            raise IIBError(
+                f'The exported package {package_name} is invalid: '
+                f'{", ".join(output.validation_dict["errors"])}'
+            )
+    except errors.OpCourierBadBundle as err:
+        raise IIBError(f'The exported package {package_name} is invalid: {err}')
+
+
+def _zip_package(package_dir):
+    """
+    Zip content of exported package to a `manifests.zip` file
+
+    :param str package_dir: path to the exported package directory
+    :raises iib.exceptions.IIBError: if unable to zip the exported package
+    """
+    base_dir, package_name = package_dir.split('/download-', 1)
+    try:
+        shutil.make_archive(f'{base_dir}/manifests', 'zip', package_dir)
+    except Exception as err:
+        raise IIBError(f'Unable to zip exported package for {package_name}: {err}')
+
+
+def _push_package_manifest(package_dir, cnr_token, organization):
+    """
+    Push `manifests.zip` file created for an exported package to OMPS
+
+    :param str package_dir: path to the exported package directory
+    :param str cnr_token: the token required to push back ported packages to the legacy
+        app registry via OMPS.
+    :param str organization: organization name in the legacy app registry to which the back ported
+        packages should be pushed to.
+    :raises iib.exceptions.IIBError: if the push is unsucessful
+    """
+    conf = get_worker_config()
+    base_dir = package_dir.split('/download-', 1)[0]
+    # Taken from https://url.corp.redhat.com/pub-code
+    with open(f'{base_dir}/manifests.zip', 'rb') as fobj:
+        files = {'file': (fobj.name, fobj)}
+        log.info('Files are %s', files)
+        resp = requests.post(
+            conf['iib_omps_url'].format(organization=organization),
+            headers={'Authorization': cnr_token},
+            files=files,
+        )
+        if not resp.ok:
+            log.error('Request to OMPS failed: %s', resp.text)
+            raise IIBError(f'Push to {organization} in the legacy app registry was unsucessful')
+        resp.raise_for_status()
+
+
+def _get_rebuilt_index_image(request_id):
```

Comment:
Since this duplicates what's done in `_create_and_push_manifest_list`, I think you should either make this a public function that is used by both or just not bother with this private function at all since it's only called one time.